sharding/database/database.go

```diff
@@ -7,11 +7,36 @@ import (
 	"path/filepath"
 
 	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
 )
 
+type ShardDB struct {
+	db *ethdb.LDBDatabase
+}
+
 // NewShardDB initializes a shardDB that writes to local disk.
-func NewShardDB(dataDir string, name string) (ethdb.Database, error) {
+func NewShardDB(dataDir string, name string) (*ShardDB, error) {
 	// Uses default cache and handles values.
 	// TODO: allow these arguments to be set based on cli context.
-	return ethdb.NewLDBDatabase(filepath.Join(dataDir, name), 16, 16)
+
+	db, err := ethdb.NewLDBDatabase(filepath.Join(dataDir, name), 16, 16)
```

Comment:
I think this needs to be part of the Start function. Does this initiate connections or anything?

Reply:
`NewLDBDatabase` opens the db and check for errors and abort if opening the db failed. You are right. It makes most sense to be under `Start()`